libs/ui/src/hooks/smartcard_auth/use_dipped_smartcard_auth.ts

```diff
@@ -170,12 +173,14 @@ function smartcardAuthReducer(scope: DippedSmartcardAuthScope) {
 
       case 'check_passcode': {
         assert(previousState.auth.status === 'checking_passcode');
+        const isPasscodeCorrect = areAllZeroSmartcardPinsEnabled()
+          ? action.passcode === '000000'
+          : action.passcode === previousState.auth.user.passcode;
```

Comment:
why is it necessary to flag here as well if the card itself has an all-zero PIN?

Reply:
Good q! My thinking was that, whenever the flag is disabled, you may create cards with real non-zero PINs. When this flag is enabled, you should ideally be able to get into any machine, even using those cards with real non-zero PINs, by entering 000000. But I realize that this adds complexity.

@mattroe Given the use cases you're imagining, any thoughts here? Is it useful / important to be able to get into machines using cards with real PINs generated when the flag was not enabled by entering 000000?

Reply:
@arsalansufi It's OK if one needs to program a card on VxAdmin with this flag set before being able to login with 000000. I think it's pretty unlikely that cards programmed with random PINs would be used in this circumstance as we are now enforcing EM-card election-hash checking that forces one to program for the given election definition.

Overall, it's unlikely that I would need to login as 000000 for a random PIN card.

Reply:
Awesome, thank you for the fresh eyes @jonahkagan and thank you for confirming @mattroe!

> I think it's pretty unlikely that cards programmed with random PINs would be used in this circumstance as we are now enforcing EM-card election-hash checking

This is a great point. Removed the modified PIN checking logic and kept only the modified PIN generation logic ✅